### DISCUSSION TITLE
Error response from daemon: could not select device driver "nvidia" with capabilities: [[gpu]]

### DISCUSSION THREAD

**Discussion by rajinder-yadav** [open-webui/open-webui#323] [General]
# Bug Report

## Description

Getting error running with gpu support.

**Bug Summary:**

The following error is report, 

```sh
Error response from daemon: could not select device driver "nvidia" with capabilities: [[gpu]]
```

**Steps to Reproduce:**

User the docker compose command from README

**Expected Behavior:**

Expecting to run with NVidia GPU support.

**Actual Behavior:**
[Describe what actually happened.]

## Environment

Operating System: openSUSE Tumbleweed 20231218
KDE Plasma Version: 5.27.10
KDE Frameworks Version: 5.113.0
Qt Version: 5.15.11
Kernel Version: 6.6.6-1-default (64-bit)
Graphics Platform: X11
Processors: 24 × AMD Ryzen 9 5900X 12-Core Processor
Memory: 62.7 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3080/PCIe/SSE2
Manufacturer: Micro-Star International Co., Ltd.
Product Name: MS-7A38
System Version: 8.0

## Reproduction Details

**Confirmation:**
- [x] I have read and followed all the instructions provided in the README.md.
- [ ] I have reviewed the troubleshooting.md document.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by dnviti** [comment]
It is necessary to install the nvidia container drivers. https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html

**Comment by rajinder-yadav** [comment]
Thanks @dnviti

**Comment by rajinder-yadav** [comment]
For openSUSE TW, I am getting a dependency error:

```sh
sudo zypper --gpg-auto-import-keys install -y nvidia-container-toolkit

Problem: nothing provides 'libseccomp' needed by the to be installed nvidia-container-toolkit-1.15.0~rc.1-1.x86_64
 Solution 1: do not install nvidia-container-toolkit-1.15.0~rc.1-1.x86_64
 Solution 2: break nvidia-container-toolkit-1.15.0~rc.1-1.x86_64 by ignoring some of its dependencies
```

The only thing I managed to find for this dependecy is "libseccomp2" which I installed using:

```sh
sudo zypper install libseccomp2
```

However this doesn't fix the initiall install error.

**Reply by dnviti** [reply]
Unfortunately i don't use openSUSE and i cannot support it.
The only thing i can guarantee is that RedHat and Debian based distros work as intended.

**Reply by rajinder-yadav** [reply]
This is an issue with NVidia and openSUSE package manager. I just put this up here incase someone else comes about with openSUSE, hopefully someone might post a solution.

[UPDATE]: Looks like there is an open issue:

https://github.com/NVIDIA/nvidia-container-toolkit/issues/110

**Comment by trharmon0** [comment]
I'm getting an identical error message with:

```./run-compose.sh --enable-gpu```

but using an AMD Radeon. I assume, like with the NVIDIAs, there's probably an additional package to install and/or a `docker build` which needs to happen?
